trustfall_core/src/interpreter/hints/tests/mod.rs

```diff
@@ -580,6 +580,31 @@ mod static_property_values {
         assert_eq!(adapter.on_starting_vertices.borrow()[&vid(1)].calls, 1);
     }
 
+    #[test]
+    fn required_properties_test() {
+        let input_name = "required_properties";
+
+        let adapter = TestAdapter {
+            on_starting_vertices: btreemap! {
+                vid(1) => TrackCalls::<ResolveInfoFn>::new_underlying(Box::new(|info| {
+                    assert!(info.coerced_to_type().is_none());
+                    assert_eq!(vid(1), info.vid());
+
+                    assert_eq!(vec![RequiredProperty::new("value".into()),
+                                    RequiredProperty::new("multiple".into()),
+                                    RequiredProperty::new("predecessor".into()),
+                                    RequiredProperty::new("__typename".into())
+                                ], info.required_properties());
```

Comment:
Heads-up that this check might break if the order of data returned by `required_properties()` ever changes.

This is a great thing to consider and then explicitly document in the doc comment of that trait method.